mobile/test/trade_test.dart

```diff
@@ -0,0 +1,114 @@
+      routerConfig: _router,
+      theme: ThemeData(
+        primarySwatch: Colors.blue,
+        extensions: const <ThemeExtension<dynamic>>[
+          // Need the trade theme otherwise the trade widgets that rely on it can't find it on the context and fail to render
+          TradeTheme(),
+        ],
+      ),
+    );
+  }
+}
+
+void main() {
+  testWidgets('Buy order flow until submission', (tester) async {
+    MockOrderService orderService = MockOrderService();
+    MockTradeValuesService tradeValueService = MockTradeValuesService();
+
+    // TODO: we could make this more resilient in the underlying components...
+    // return dummies otherwise the fields won't be initialized correctly
+    when(tradeValueService.calculateMargin(
+            price: anyNamed('price'),
+            quantity: anyNamed('quantity'),
+            leverage: anyNamed('leverage')))
+        .thenReturn(Amount(2000));
+    when(tradeValueService.calculateLiquidationPrice(
+            price: anyNamed('price'),
+            leverage: anyNamed('leverage'),
+            direction: anyNamed('direction')))
+        .thenReturn(10000);
+    when(tradeValueService.calculateQuantity(
+            price: anyNamed('price'), leverage: anyNamed('leverage'), margin: anyNamed('margin')))
+        .thenReturn(100);
+
+    SubmitOrderChangeNotifier submitOrderChangeNotifier = SubmitOrderChangeNotifier(orderService);
+
+    when(orderService.submitMarketOrder(any, any, any, any))
+        .thenAnswer((_) async => ApiResponse(status: ResponseStatus.success));
+
+    await tester.pumpWidget(MultiProvider(providers: [
+      ChangeNotifierProvider(create: (context) => TradeValuesChangeNotifier(tradeValueService)),
+      ChangeNotifierProvider(create: (context) => submitOrderChangeNotifier),
+      ChangeNotifierProvider(create: (context) => OrderChangeNotifier(orderService)),
+      ChangeNotifierProvider(create: (context) => AmountDenominationChangeNotifier()),
+    ], child: const TestWrapperWithTradeTheme(child: TradeScreen())));
+
+    await tester.pumpAndSettle();
+
+    expect(find.byKey(tradeScreenButtonBuy), findsOneWidget);
+
+    // Open bottom sheet
+    await tester.tap(find.byKey(tradeScreenButtonBuy));
+    await tester.pumpAndSettle();
+
+    expect(find.byKey(tradeScreenBottomSheetButtonBuy), findsOneWidget);
+
+    // click buy button in bottom sheet
+    await tester.tap(find.byKey(tradeScreenBottomSheetButtonBuy));
+    await tester.pumpAndSettle();
+
+    expect(find.byKey(tradeScreenBottomSheetConfirmationSliderButtonBuy), findsOneWidget);
+
+    // TODO: This is not optimal because if we re-style the component this test will likely break.
+    // Drag to confirm
+    await tester.timedDrag(find.byKey(tradeScreenBottomSheetConfirmationSliderButtonBuy),
+        const Offset(275, 0), const Duration(seconds: 2),
+        pointer: 7);
+
+    verify(orderService.submitMarketOrder(any, any, any, any)).called(1);
```

Comment:
Should we verify that `submitMarketOrder` was submitted with the correct values?

Reply:
I had that in mind at some point, but I would probably test that in a different test because it's not the concern here.

This test basically clicks through the flow to ensure that the models pop up correctly and we trigger submitting the order after confirmation.
I would see a smaller unit test in charge of checking on the actual values, unless we fear that the values might be wrong in the end because of something happening in the flow.

Reply:
> Looks good. 2 points:
> 
>     * I think we can iterate on the test itself later if you want

sweet

>     * it would be good to execute these tests in CI aswell.

Let's give this a try. If it's not as trivial as I think it is I will post here and do a follow up PR :)

Reply:
![image](https://user-images.githubusercontent.com/5557790/219557385-733a86b2-f068-4240-9e03-72068a799819.png)

easy :)